analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CheckArgumentException.CSharp9.cs

```diff
@@ -116,6 +116,16 @@ public record TestRecord()
     }
 }
 
+public record R(string Prop)
+{
+    public class TestClass
+    {
+        public static string ClassProperty { get; }
+
+        public string RecordProperty { get; } = ClassProperty ?? throw new ArgumentNullException(nameof(Prop)); // Compliant - Prop is a record parameter
```

Comment:
I consider this case noncompliant since the `Prop` is not available for use in the nested class. The following code snippet will not compile.
`public string RecordProperty { get; } = Prop ?? throw new ArgumentNullException(nameof(Prop));`